dds/DCPS/DiscoveryBase.h

```diff
@@ -1263,6 +1231,244 @@ namespace OpenDDS {
         }
       }
 
+      void
+      remove_expired_endpoints(const DCPS::MonotonicTimePoint& /*now*/)
+      {
+        // TLS_TODO: sanity check on locks
+        ACE_GUARD(ACE_Thread_Mutex, g1, matching_data_buffer_lock_);
+        MatchingDataIter it;
+        for (it = matching_data_buffer_.begin(); it != matching_data_buffer_.end(); it++) {
+          if (MonotonicTimePoint::now() - it->second.time_added_to_map >= max_type_lookup_service_reply_period_) {
+            matching_data_buffer_.erase(it->first);
+          }
+        }
+        // TLS_TODO: any other cleanup?
```

Comment:
I don't think so. Besides anything you're doing I don't think there's anything to cleanup, because all the action is in `match_continue`.

Reply:
I am more concerned about possible structures that existed prior to this project. For example, when a new endpoint has been discovered and may have been added to some sort of 'structure' for further processing, but now does not reply to the type lookup service request and is essentially 'useless'. Do such 'structures' exist, and if so should the discovered endpoint be removed from them?

Reply:
No, I don't think so. I would only expect things like that to happen for builtin endpoints. Regular entities shouldn't do anything until they've gotten the go ahead that they've been matched.